hazelcast-jet-core/src/main/java/com/hazelcast/jet/Outbox.java

```diff
@@ -58,20 +58,20 @@ default void add(@Nonnull Object item) {
 
     /**
      * Returns {@code true} if any of this outbox's buckets is above its
```

Comment:
Yes, but we should document, that the call fails in case of full outbox. Assertions are disabled by default and some might not notice, until after they enable them.

I would just throw `JetException`. Assertions should be used to check for our programming errors, not for user errors. This assertion is easily violated by the user.